src/bootstrap/doc.rs

```diff
@@ -707,7 +713,7 @@ macro_rules! tool_doc {
                     target,
                     "doc",
                     $path,
-                    SourceType::InTree,
+                    source_type,
                     &[],
                 );
```

Comment:
If this command fails for Miri, that failure must be ignored. Miri failing to build is not a bug, it is expected. That's what the entire toolstate stuff is about.